results/pkg/watcher/reconciler/controller_test.go

```diff
@@ -157,4 +225,39 @@ func TestReconcile(t *testing.T) {
 			t.Fatalf("Expected completed TaskRun should be upated in api server: %v", diff)
 		}
 	})
+
+	// Update a PipelineRun with new value and results id, completed PipelineRun should be updated.
+	t.Run("test-update-pipelinerun", func(t *testing.T) {
+		pr, err := reconcilePipelineRun(ctx, asset, pipelineRun)
+		if err != nil {
+			t.Fatalf("Failed to get completed PipelineRun %s: %v", pipelineRun.Name, err)
+		}
+		pr.UID = "234435"
+		asset.Clients.Pipeline.TektonV1beta1().PipelineRuns(pipelineRun.Namespace).Update(pr)
+		updatepr, err := reconcilePipelineRun(ctx, asset, pr)
+		updatepr.ResourceVersion = pr.ResourceVersion
+		if err != nil {
+			t.Fatalf("Failed to get completed PipelineRun %s: %v", pipelineRun.Name, err)
+		}
+		if diff := cmp.Diff(pr, updatepr); diff != "" {
+			t.Fatalf("Expected completed PipelineRun should be updated in cluster: %v", diff)
+		}
+		res, err := client.GetResult(ctx, &pb.GetResultRequest{Name: pr.Annotations[IDName]})
+		if err != nil {
+			t.Fatalf("Expected completed PipelineRun %s not created in api server", pipelineRun.Name)
+		}
+		p, err := convert.ToPipelineRunProto(updatepr)
+		if err != nil {
+			t.Fatalf("failed to convert to proto: %v", err)
+		}
+		want := &pb.Result{
+			Name: pr.Annotations[IDName],
+			Executions: []*pb.Execution{{
+				Execution: &pb.Execution_PipelineRun{p},
+			}},
+		}
+		if diff := cmp.Diff(want, res, protocmp.Transform()); diff != "" {
+			t.Fatalf("Expected completed PipelineRun should be upated in api server: %v", diff)
+		}
+	})
 }
```

Comment:
Another test that would be useful is an additive Result - e.g. have a PipelineRun, then add a TaskRun in the same result. As a bonus, let's have them share the same name so we can make sure to cover that edge case as well.

Reply:
I have a question here. If we separate the reconcilers, is there any chance we would meet this corner case?

Reply:
We wouldn't have to worry about the naming corner case, but we should make sure to test the additive behavior (e.g. TaskRuns and PipelineRuns are able to be added to the same Result)

Reply:
Oh! That's a great Test Case!

I didn't get that you mean we should be able to hold both PipelineRun and TaskRun in the same result, I thought you want to check whether it would work if they share the same name.